framework/src/transfers/MultiAppGeneralFieldShapeEvaluationTransfer.C

```diff
@@ -125,7 +125,9 @@ MultiAppGeneralFieldShapeEvaluationTransfer::evaluateInterpValuesWithMeshFunctio
       else
       {
         // Use mesh function to compute interpolation values
-        auto val = (local_meshfuns[i_from])(pt - _from_positions[i_from]);
+        const auto from_global_num =
+            _current_direction == TO_MULTIAPP ? 0 : _from_local2global_map[i_from];
```

Comment:
Are you going to use the new methods you made?

Reply:
I added them after! I ll do it now